image-customize

```diff
@@ -82,6 +83,114 @@ class InstallAction(ActionBase):
         machine_instance.execute(f"{install_command} {package}", timeout=1800)
 
 
+class BuildAction(ActionBase):
+    '''Build and install distribution package(s) from dist tarball'''
+
+    quick = False
+    verbose = False
+
+    @staticmethod
+    def execute(machine_instance, tarball):
+        # upload the tarball
+        tarname = os.path.basename(tarball)
+        vm_tar = os.path.join("/var/tmp", tarname)
+        # looks like cockpit-foo-123.10.gdbde381.tar.xz
+        m = re.search(r'([^-]+)\.tar(\..*)?$', tarname)
```

Comment:
The intention is "give me the last of the - separated components", which is the version number. Indeed that assumes that the version number does not contain `-`, but that assumption is true in the projects for which we use this.

What's the advantage of explicitly matching the `^.+-` prefix? We don't use it for anything, and we can't even assume that it starts with cockpit (starter-kit!)

Reply:
`fullmatch()` just seems like a better fit than `search()` here.  We're not really *searching* per se.  We have a thing which has a precisely known format.  Let's describe that format.

It also specifically allows to avoid the `^` and `$` anchors, as they become implicit.

At the very least please add a `-` to the front here to show that a `-` should immediately precede the version number.

Reply:
Changed to fullmatch, but honestly I find that *less* readable/comprehensible. (But then again, no strong opinion).